resolver/master_changer.go

```diff
@@ -0,0 +1,133 @@
+package resolver
+
+import (
+	"encoding/binary"
+	"net"
+	"sort"
+	"strconv"
+
+	mesos "github.com/mesos/mesos-go/mesosproto"
+
+	"github.com/mesosphere/mesos-dns/logging"
+)
+
+// MasterChanger stores the current state
+// of the current leader/masters
+type MasterChanger struct {
+	// current masters list,
+	// 1st item represents the leader,
+	// the rest remaining masters
+	masters []string
+
+	changed chan []string
+}
+
+// NewMasterChanger returns a new MasterChanger
+// with an unknown (empty) leader, the given masters
+// and the initial channel capacity for emitting changes
+func NewMasterChanger(masters []string, cap int) *MasterChanger {
```

Comment:
Well, since you already changed the interface here to account for capacity and not have to run `Emit` in a go-routine, couldn't we go all the way and implement the `Subscribe` method and get rid of the `ChangedChan` and `Emit`?